endpoints.go

```diff
@@ -144,9 +176,18 @@ var (
 	EndpointEmoji         = func(eID string) string { return EndpointCDN + "emojis/" + eID + ".png" }
 	EndpointEmojiAnimated = func(eID string) string { return EndpointCDN + "emojis/" + eID + ".gif" }
 
-	EndpointOauth2            = EndpointAPI + "oauth2/"
-	EndpointApplications      = EndpointOauth2 + "applications"
-	EndpointApplication       = func(aID string) string { return EndpointApplications + "/" + aID }
-	EndpointApplicationsBot   = func(aID string) string { return EndpointApplications + "/" + aID + "/bot" }
-	EndpointApplicationAssets = func(aID string) string { return EndpointApplications + "/" + aID + "/assets" }
+	EndpointApplications = EndpointAPI + "applications"
```

Comment:
I needed to change it. Because EndpointApplications now doesn't belong to OAuth2. It belongs to applications and their commands.

Reply:
Yeah but this might cause a problem on the off chance someone was relying on EndpointApplications to be with OAuth2.
It IS an exported variable after all.

Reply:
> Yeah but this might cause a problem on the off chance someone was relying on EndpointApplications to be with OAuth2.
> It IS an exported variable after all.

Reasonable question: how I need to name it then, tell me. Applications and OAuth2 applications isn't the same thing now. And also, why are everyone tries to save backward compatibility even when WE JUST CAN'T???? Because API is changed and we need to change something, not sit on the same place as we was before.

Reply:
Adding things to an API doesn't break compatibility, but removing and modifying things can eg:

For example, something as innocuous as:

```
fmt.Println(discordgo.EndpointApplication)
```

This will now cause a compiler error when updating with this PR.

It's our job as library maintainers to minimize these scenarios, even when they cause the code to not be exactly what we would like in a perfect world.

Ps. It feels like you're getting upset by this code review, please remember that these comments aren't a criticism of you as a person, just they reflect the combined experience of a number of people also trying to do a good job - we're working together to make this the best PR possible!

Reply:
> It feels like you're getting upset by this code review
Of course I'm upset.

> but removing and modifying things
I can't save `EndpointApplication` here, because Discord API changed it and it no longer "OAuth2" applications. Or at least,  I'm waiting for suggestions what name I should pick for new ones if I will return EndpointApplications and other OAuth2 stuff back to original.

Reply:
Code review isn't a criticism of you as a person, I'm sorry you're feeling upset, but that is not the intention. As the maintainers of the library, we have a responsibility to our users, and we are essentially signing up to being the maintainers of this code in the future.

My apologies for speaking without fully getting context on this.

I've done some searching, it seems like there is no code on Github or Google currently using EndpointApplication, I'm happy with renaming this to EndpointOauth2Application, but I would like us to add a warning in the release notes.

Reply:
Yeah, that's right, I appreciate that decision. Well, actually, we can try to create another name for current applications endpoint, but... that would be really hard to rename in future. So, yeah, you're right, warning in release notes is enough for this. Btw, when will be release?

Reply:
>I've done some searching, it seems like there is no code on Github or Google currently using EndpointApplication

Just want to note that many of the larger discord bots tend to be private codebases, and those may be more likely to use oauth flows in e.g. web dashboards or what not, where these variables may be used @iopred